lib/dtutils/processor.lua

```diff
@@ -0,0 +1,266 @@
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local dtutils_processor = {}
+
+dtutils_processor.libdoc = {
+  Sections = {"Name", "Synopsis", "Description", "License"},
+  Name = [[dtutils.processor - Darktable lua functions for building processor scripts]],
+  Synopsis = [[local dp = require "lib/dtutils.processor"]],
+  Description = [[dtutils.processor provides common functions used for building scripts
+    that send images out to an external process, process them, and return the result (i.e. processors).]],
+  License = [[This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.]],
+  functions = {}
+}
+
+local dt = require "darktable"
+local dtutils = require "lib/dtutils"
+local df = require "lib/dtutils.file"
+local log = require "lib/libLog"
+
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+-- This should be $HOME/.config/darktable/lua/locale/
+gettext.bindtextdomain("dtutils",dt.configuration.config_dir.."/lua/locale/")
+
+local function _(msgid)
+    return gettext.dgettext("dtutils.processor", msgid)
+end
+
+--[[
+  NAME
+    exporter_status - show the status while exporting images
+
+  SYNOPSIS
+    local dp = require "lib/dtutils.processor"
+
+    local result = dp.exporter_status(storage, image, format, filename, number, total, high_quality, extra_data)
+      storage - dt_imageio_module_storage_t - the storage that the status is being shown for
+      image - dt_lua_image_t - the current image
+      format - dt_imageio_module_format_t - the format of the export image
+      filename - string - the filename being exported to
+      number - integer - the number of this image in the sequence
+      total - integer - the total number of images being exported
+      high_quality - boolean
+      extra_data - extra data from the storage
+
+  DESCRIPTION
+    exporter_status runs prior to each image being exported and prints out a status that lets the user
+    know how many images, out of the total, have been exported
```

Comment:
"should be run" (unless it's an internal function to dtutils.processor)

also, the important part is that it's printing the result in a common way. you might want to show what it prints in your example so people understand better what it does